config/src/test/java/tech/pegasys/pantheon/config/CliqueConfigOptionsTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.config;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+
+import io.vertx.core.json.JsonObject;
+import org.junit.Test;
+
+public class CliqueConfigOptionsTest {
+
+  private static final long EXPECTED_DEFAULT_EPOCH_LENGTH = 30_000;
+  private static final int EXPECTED_DEFAULT_BLOCK_PERIOD = 15;
+
+  @Test
+  public void shouldGetEpochLengthFromConfig() {
+    final CliqueConfigOptions config = fromConfigOptions(singletonMap("epochLength", 10_000));
```

Comment:
So - does this imply you an create a valid clique config options,even if period is missing?

Reply:
Yes, if it's not specified it would use the default.  But mostly it just means `CliqueConfigOptions` isn't doing validation and I'm taking advantage of that to keep the test as simple as possible.